### PR TITLE
cmd/devp2p: fixes for eth and discv4 tests

### PR DESCRIPTION
**Pull request by holiman**
This PR fixes a false positive ` PONG 'to' endpoint mismatch: got {IP:172.17.0.7 UDP:44025 TCP:44025}, want {IP:172.17.0.7 UDP:44025 TCP:0}` in the hive tests, and adds some packet decoding tests.

### CODE REVIEW

File: p2p/discover/v4wire/packet_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+import (
+	"encoding/hex"
+	"testing"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/ethereum/go-ethereum/rlp"
+)
+
+func TestDecode(t *testing.T) {
```
Comment (fjl) [inline p2p/discover/v4wire/packet_test.go:27]: I'm not sure we should keep this test. The test basically checks that RLP decoding works, but there are enough tests in package rlp to cover this, I think.

Comment (holiman) [reply p2p/discover/v4wire/packet_test.go:27]: Yeah, sure

**Comment by fjl** [comment]
@holiman internal tests are failing now, so we can't merge it yet.

**Comment by holiman** [comment]
Failures fixed now, ptal @fjl

**Pull request merged**